List Ops: undefined func

https://github.com/exercism/go/blob/8de5b265ce06df2057e1c4d7e0a0bd9a2fb92ba6/exercises/practice/list-ops/list_ops_test.go#L11

```sh
# listops [listops.test]
./list_ops_test.go:11:11: undefined: binFunc
FAIL	listops [build failed]
'/usr/local/go/bin/go test --short --json .' returned exit code 2: exit status 2
```

--- a/exercises/practice/list-ops/.meta/example.go
+++ b/exercises/practice/list-ops/.meta/example.go
@@ -2,12 +2,9 @@ package listops
 
 // IntList is an abstraction of a list of integers
 type IntList []int
-type predFunc func(int) bool
-type binFunc func(int, int) int
-type unaryFunc func(int) int
 
 // Foldl folds (reduces) the given list from the left with a function
-func (s IntList) Foldl(fn binFunc, initial int) int {
+func (s IntList) Foldl(fn func(int, int) int, initial int) int {
 	if len(s) == 0 {
 		return initial
 	}
@@ -16,17 +13,17 @@ func (s IntList) Foldl(fn binFunc, initial int) int {
 }
 
 // Foldr folds (reduces) the given list from the right with a function
-func (s IntList) Foldr(fn binFunc, initial int) int {
+func (s IntList) Foldr(fn func(int, int) int, initial int) int {
 	flippedFunc := func(x, y int) int { return fn(y, x) }
 	// Note: This relies on s being finite
 	return s.Reverse().Foldl(flippedFunc, initial)
 }
 
 // Filter list returning only values that satisfy the filter function
-func (s IntList) Filter(fn predFunc) IntList {
+func (s IntList) Filter(fn func(int) bool) IntList {
 	filtered := make([]int, 0, len(s))
-	var filterAcc func(predFunc, IntList, IntList) IntList
-	filterAcc = func(fn predFunc, acc IntList, lst IntList) IntList {
+	var filterAcc func(func(int) bool, IntList, IntList) IntList
+	filterAcc = func(fn func(int) bool, acc IntList, lst IntList) IntList {
 		if len(lst) == 0 {
 			return acc
 		}
@@ -43,7 +40,7 @@ func (s IntList) Filter(fn predFunc) IntList {
 func (s IntList) Length() int { return len(s) }
 
 // Map returns a list of elements whose values equal the list value transformed by the mapping function
-func (s IntList) Map(fn unaryFunc) IntList {
+func (s IntList) Map(fn func(int) int) IntList {
 	newSlice := make([]int, len(s))
 	for idx, elt := range s {
 		newSlice[idx] = fn(elt)

--- a/exercises/practice/list-ops/list_ops.go
+++ b/exercises/practice/list-ops/list_ops.go
@@ -3,7 +3,7 @@ package listops
 // IntList is an abstraction of a list of integers which we can define methods on
 type IntList []int
 
-func (s IntList) Foldl(fn func(int) bool, initial int) int {
+func (s IntList) Foldl(fn func(int, int) int, initial int) int {
 	panic("Please implement the Foldl function")
 }
 

--- a/exercises/practice/list-ops/list_ops_test.go
+++ b/exercises/practice/list-ops/list_ops_test.go
@@ -8,7 +8,7 @@ import (
 var foldTestCases = []struct {
 	name     string
 	property string
-	fn       binFunc
+	fn       func(int, int) int
 	initial  int
 	list     IntList
 	want     int
@@ -84,7 +84,7 @@ func TestFold(t *testing.T) {
 var filterTestCases = []struct {
 	name     string
 	property string
-	fn       predFunc
+	fn       func(int) bool
 	list     []int
 	want     []int
 }{
@@ -153,7 +153,7 @@ var mapTestCases = []struct {
 	name     string
 	property string
 	list     IntList
-	fn       unaryFunc
+	fn       func(int) int
 	want     IntList
 }{
 	{

